### DISCUSSION TITLE
Information/documentation about Wahoo Direct Connect

### DISCUSSION THREAD

**Discussion by VincentD75** [cagnulein/qdomyos-zwift#1686] [Q&A]
Hello,

I would like to play with Wahoo Direct Connect and wondering if you have some documentation to share with me ?

thank you.

**Comment by cagnulein** [comment]
Hi, no unfortunately there is no documentation at all. We have reverse
engineered it. Which is your target?

Il giorno mar 3 ott 2023 alle 15:26 VincentD75 ***@***.***>
ha scritto:

**Comment by VincentD75** [comment]
Hello Roberto,

I would like to create a small application to decrease power from my kickr
bike.
It is reading 5% too high.

Le mar. 3 oct. 2023 à 15:29, Roberto Viola ***@***.***> a
écrit :

**Comment by cagnulein** [comment]
why don't you simply use qz? it's already doing this using the watt gain
setting

Il giorno mar 3 ott 2023 alle 23:12 VincentD75 ***@***.***>
ha scritto:

**Reply by kopierreko** [reply]
Hi @cagnulein. I'm searching for a way to get data from Direct Connect and drive the resistor. Do you have a documentation of your reverse engineering ?
Thanks 😊 
PS: I don't want to make another cycling software, I'm doing a "game" IRL with machines and I want to use the wahoo home trainer to display datas.

**Reply by cagnulein** [reply]
no, as I said I don't have any documentation.

**Reply by kopierreko** [reply]
Okay 😊 But is it possible to re-use your work to use the ethernet connexion with the home trainer ?

**Reply by cagnulein** [reply]
QZ implemented the opposite way :) If you want I can implement the ethernet connection with the home trainer. Do you think you can add your "IRL game" into QZ? Could you please describe it more?

**Reply by kopierreko** [reply]
I work with video-mapping and mechanisms so I "just" want to "convert" the kickr home trainer communication to OSC to have lot of flexibility

**Reply by cagnulein** [reply]
I worked on the video-mapping field my self too in the last 10 years :) I still do not understand why you would like to use the kickr protocol, but anyway it will be a long task, if you are doing it as a job, maybe you can contact me in private and I can do it as a real task. Let me know

**Reply by kopierreko** [reply]
I can describe the project, how can I contact you ?

**Reply by cagnulein** [reply]
roberto.viola83 at gmail.com @kopierreko
